samples/gen-go/client/doer.go

```diff
@@ -91,3 +99,196 @@ func (d retryDoer) Do(c *http.Client, r *http.Request) (*http.Response, error) {
+		"errorCount":                      e.ErrorCount,
+		"errorPercentage":                 e.ErrorPercentage,
+		"isCircuitBreakerOpen":            e.IsCircuitBreakerOpen,
+		"rollingCountFailure":             e.RollingCountFailure,
+		"rollingCountFallbackFailure":     e.RollingCountFallbackFailure,
+		"rollingCountFallbackSuccess":     e.RollingCountFallbackSuccess,
+		"rollingCountShortCircuited":      e.RollingCountShortCircuited,
+		"rollingCountSuccess":             e.RollingCountSuccess,
+		"rollingCountThreadPoolRejected":  e.RollingCountThreadPoolRejected,
+		"rollingCountTimeout":             e.RollingCountTimeout,
+		"currentConcurrentExecutionCount": e.CurrentConcurrentExecutionCount,
+		"latencyTotalMean":                e.LatencyTotalMean,
+	})
+
+}
+
+func (d *circuitBreakerDoer) init() {
+	// Turn on the SSE port + listener once
+	circuitSSEOnce.Do(func() {
+		hystrixStreamHandler := hystrix.NewStreamHandler()
+		hystrixStreamHandler.Start()
+		go http.ListenAndServe(":8111", hystrixStreamHandler)
+		go func() {
+			logger := logger.New("wag")
+			logFrequency := 30 * time.Second
+			lastEventSeen := map[string]HystrixSSEEvent{}
+			lastEventLogTime := map[string]time.Time{}
+
+			for _ = range time.Tick(1 * time.Second) { // retry indefinitely
+				stream, err := eventsource.Subscribe("http://127.0.0.1:8111", "")
+				if err != nil {
+					log.Printf("error connecting to circuit SSE stream: %s", err)
+					continue
+				}
+				for ev := range stream.Events {
+					var e HystrixSSEEvent
+					if err := json.Unmarshal([]byte(ev.Data()), &e); err != nil {
+						continue
+					}
+					if e.Type != "HystrixCommand" {
+						continue
+					}
+					lastSeen, ok := lastEventSeen[e.Name]
+					lastEventSeen[e.Name] = e
+
+					// log the circuit state if it's either
+					// (1) the first event we've seen for the circuit or
+					// (2) the circuit open state has changed or
+					// (3) 30 seconds have passed since we logged something for the circuit
+					if !ok {
+						lastEventLogTime[e.Name] = time.Now()
+						logEvent(logger, e)
+						continue
+					}
+					if lastSeen.IsCircuitBreakerOpen != e.IsCircuitBreakerOpen {
+						lastEventLogTime[e.Name] = time.Now()
+						logEvent(logger, e)
+					} else if time.Now().Sub(lastEventLogTime[e.Name]) > logFrequency {
+						lastEventLogTime[e.Name] = time.Now()
+						logEvent(logger, e)
+					}
+				}
+			}
+		}()
+	})
+
+	// Configure the circuit settings
+	hystrix.ConfigureCommand(d.circuitName, hystrix.CommandConfig{
+		// redundant, so need to keep this in sync with client timeout or set it high so that it's irrelevant :-/
+		Timeout: 100 * 1000,
+		// how many commands of the same type can run at the same time
+		MaxConcurrentRequests: 100,
+		// minimum number of requests needed before a circuit can be tripped due to health
+		RequestVolumeThreshold: 20,
+		// how long, in milliseconds, to wait after a circuit opens before testing for recovery
+		SleepWindow: 5000,
```

Comment:
my gut is that 5 seconds is a long time to wait to send just one request, but maybe it's better to be conservative. did we just take this number from the hysterix-go defaults?